src/providers/dataProvider/dataprovider.test.ts

```diff
@@ -0,0 +1,81 @@
+import { type DataProvider } from 'react-admin'
+import { getDataProvider } from '.'
+import { R_BATCHES } from '../../constants'
+import { DateTime } from 'luxon'
+
+const year: number = 2023
+
+const generateDummyBatchForTesting = (name?: string): Omit<Batch, 'id'> => {
+  return {
+    name: name ?? 'Dummy-Batch',
+    startDate: DateTime.now().toFormat('yyyy-MM-dd'),
+    endDate: DateTime.now().plus({ day: 1 }).toFormat('yyyy-MM-dd'),
+    projectCode: '1',
+    batchNumber: 'V01/2023',
+    yearOfReceipt: String(year),
+    department: 1,
+    project: 1,
+    platform: 1,
+    organisation: 1,
+    maximumProtectiveMarking: 1,
+    remarks: 'remarks-1',
+    receiptNotes: 'receipt-notes-1',
+    createdAt: DateTime.now().toFormat('yyyy-MM-dd'),
+    createdBy: 1
+  }
+}
+
+describe('CRUD operations on each resource', () => {
+  let provider: DataProvider
+  beforeAll(async () => {
+    provider = await getDataProvider(false)
+  })
+
+  // id is omitted because it is automatically generated by localforage while inserting
+  describe('CRUD Operation on Batch', () => {
+    let id: number
+    let batch: Batch
+
+    it('should create a batch', async () => {
+      const data = generateDummyBatchForTesting()
+      const result = await provider.create<Batch>(R_BATCHES, {
+        data
+      })
+      id = result.data.id
+      batch = result.data
+      expect(id).toBeDefined()
+      expect(batch).toBeDefined()
+    })
+
+    it('should get the batch', async () => {
+      const data = generateDummyBatchForTesting()
+      const result = await provider.getOne<Batch>(R_BATCHES, { id })
+      const { id: omittedId, ...expectedResult } = result.data
+      expect(expectedResult).toMatchObject(data)
```

Comment:
note: I don't think we should be omitting the `id` when testing `getOne`. It should be the `id` we used for `getOne`.